### ISSUE TITLE
Does NumCpp support broadcast?

### ISSUE THREAD

**Issue by for-just-we** [dpilger26/NumCpp#174]
I did subtract operation between a vector1 with shape `[2, 47]` and vector2 with shape `[1, 47]`. The error message is:

```cpp
Function: operator-
Line: 412
Error: Array dimensions do not match.libc++abi: terminating with uncaught exception of type std::invalid_argument: File: xxx/NumCpp/include/NumCpp/NdArray/NdArrayOperators.hpp
Function: operator-
Line: 412
Error: Array dimensions do not match.
```

It seems that NumCpp could not automatically transform vector2 into shape `[2, 47]`. How could I do this manually?

**Comment by dpilger26** [comment]
No, at the moment the only broadcasting supported is scalar.

**Comment by dpilger26** [comment]
Fixed in Release Version 2.10.0.
